src/compiler/scala/tools/nsc/typechecker/Typers.scala

```diff
@@ -3935,11 +3934,25 @@ trait Typers extends Adaptations with Tags with TypersTracking with PatternTyper
         else
           typedFun0
       )
+      if (typedFunPart.isErroneous) return finish(ErroneousAnnotation)
+
       val treeInfo.Applied(typedFun @ Select(New(annTpt), _), _, _) = typedFunPart
-      val annType = annTpt.tpe // for a polymorphic annotation class, this type will have unbound type params (see context.undetparams)
+      val annType = annTpt.tpe // for a polymorphic annotation class, th/t10497is type will have unbound type params (see context.undetparams)
       val annTypeSym = annType.typeSymbol
       val isJava = annType != null && annTypeSym.isJavaDefined
 
+      typedFun0 match {
+        case Select(New(a), _) =>
+
+          // test/files/jvm/annotations/Test_2.scala:97 fails if I just use a.symbol.isJavaAnnotation. Seems like it might be a bug?
+          val typedFun0ExtendsAnn = a.tpe <:< AnnotationClass.tpe || a.symbol.baseClasses.exists(x => x.isJavaAnnotation)
+
+          if (!typedFun0ExtendsAnn){
+            reportAnnotationError(DoesNotExtendAnnotation(typedFun0, a.tpe.typeSymbol.initialize))
+            return finish(ErroneousAnnotation)
+          }
```

Comment:
might be good to play it safe and add a `case _ =>`, so avoid the compiler crashing in unexpected situations (maybe under some error conditions that are not currently tested).